src/libraries/System.Threading/tests/InterlockedTests.cs

```diff
@@ -302,7 +302,7 @@ public void InterlockedOr_UInt64()
             Assert.Equal(0x17755771u, value);
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsNotArm64Process))] // [ActiveIssue("https://github.com/dotnet/runtime/issues/11177")]
```

Comment:
Instead of disabling it, can we check the OS version inside the test and skip the test if we are running on OS that is known to have bugs in this?

Reply:
Look for `throw new SkipTestException` to see the pattern to use.

Reply:
`Environment.OSVersion` returns the Linux kernel version.

Reply:
Yes we should consider that. My first reaction though is to just revert - to fix the CI nuisance.

Reply:
Also the logs are confusing, but it looks like the reported failure was on `net5.0-Windows_NT-Release-arm64-CoreCLR_release-Windows.10.Arm64.Open`

https://dev.azure.com/dnceng/public/_build/results?buildId=633940&view=ms.vss-test-web.build-test-results-tab&runId=19752174&resultId=142362&paneView=debug